src/models/application.js

```diff
@@ -129,6 +132,58 @@ function getFromSelf (appId) {
   return application.get({ appId }).then(sendToApi);
 };
 
+/**
+ * @param {{app_id: string}|{app_name: string}} appIdOrName
+ * @param {string} alias
+ * @return {Promise<{appId: string, ownerId: string}>}
+ */
+async function resolveId (appIdOrName, alias) {
+  if (appIdOrName != null && alias != null) {
+    throw new Error('Only one of `--app` or `--alias` options can be specified at a time');
+  }
+
+  // -- resolve by linked app
+
+  if (appIdOrName == null) {
+    const appDetails = await AppConfig.getAppDetails({ alias });
+    return { appId: appDetails.appId, ownerId: appDetails.ownerId };
+  }
+
+  // -- resolve by app id
+
+  if (appIdOrName.app_id != null) {
+    const ownerId = await resolveOwnerId(appIdOrName.app_id);
+    if (ownerId != null) {
+      return {
+        appId: appIdOrName.app_id,
+        ownerId,
+      };
+    }
+
+    throw new Error('Application not found');
+  }
+
+  // -- resolve by app name
+
+  const summary = await getSummary({}).then(sendToApi);
+
+  const candidates = [summary.user, ...summary.organisations]
+    .flatMap((owner) => owner.applications.map((app) => ({ app, owner })))
+    .filter((candidate) => candidate.app.name === appIdOrName.app_name);
+
+  if (candidates.length === 0) {
+    throw new Error('Application not found');
+  }
+  if (candidates.length === 1) {
+    return {
+      appId: candidates[0].app.id,
+      ownerId: candidates[0].owner.id,
+    };
+  }
+
+  throw new Error(`This name refers to multiple applications:\n${candidates.map((candidate) => `${candidate.owner.name}: ${candidate.app.id}`).join('\n')}`);
```

Comment:
I would add the orga_id too as well as the name :)

Reply:
The owner name is enough here ?

Reply:
Error text is misaligned, it can be replaced by: 

```suggestion
    Logger.printErrorLine(`The name '${appIdOrName.app_name}' refers to multiple applications:`);
  candidates.forEach((candidate) => {
    Logger.printErrorLine(`- ${candidate.owner.name}: ${candidate.app.id} (${candidate.app.variantSlug})`);
  });
  throw new Error('Ambiguous application name, use the `--app` option with one of the IDs above');
}
```